PYME/Acquire/SpoolController.py

```diff
@@ -265,41 +297,49 @@ def _get_queue_name(self, fn, pcs=False):
         return self._sep.join([self.dirname.rstrip(self._sep), fn + ext])
 
 
-    def StartSpooling(self, fn=None, stack=None, compLevel=None, zDwellTime=None, doPreflightCheck=True, maxFrames=sys.maxsize,
-                      pzf_compression_settings=None, cluster_h5 = None):
-        """Start spooling
+    def StartSpooling(self, filename=None, max_frames=sys.maxsize, 
+                      do_preflight_check=True, **kwargs):
         """
+
+        Parameters
+        ----------
+        filename : str, optional
+            filename can be hardcoded here, otherwise differs to the seriesName
+            property which will create one if need-be.
+        max_frames : int, optional
+            point at which to end the series automatically, by default 
+            sys.maxsize
+        do_preflight_check : bool, optional
+            toggle performing pre-flights specified in the acquisition protocol,
+            by default True.
+        kwargs : dict
+            can be used to update the SpoolController settings at the beginning
+            of this spool. See `SpoolController.update_settings`
         
-        # these settings were managed by the GUI, but are now managed by the controller, still allow them to be passed in,
-        # but default to using our internal values
-        compLevel = self.hdf_compression_level if compLevel is None else compLevel
-        pzf_compression_settings = self.pzf_compression_settings if pzf_compression_settings is None else pzf_compression_settings
-        stack = self.z_stepped if stack is None else stack
-        cluster_h5 = self.cluster_h5 if cluster_h5 is None else cluster_h5
-        fn = self.seriesName if fn in ['', None] else fn
-        zDwellTime = self.z_dwell if zDwellTime is None else zDwellTime
+        """
+        filename = self.seriesName if filename in ['', None] else filename
+        self.update_settings(kwargs)
```

Comment:
I need to think a bit more about whether we should be updating the settings, or just over-riding with anything we supply. 

In principle I'd prefer not to update the spool controller settings - i.e. calling `StartSpooling()` with custom settings should over-ride the settings for this call but not change behaviour for subsequent calls which don't provide settings. 

The caveat here would relate to your use case of changing the spool directory for different wells, but this might be able to be handled separately.

Reply:
I think if we're in a situation where we're doing things that involve changing things between acquisitions, we are probably also in a situation where we'd better pair settings with every startspool task. Should mention I need to change the protocol too (and with chained analysis stuff hope to not need the directory change forever).

Reply:
I didn't phrase that well, but I think once we have mixed acquisition types there are no 'normal' settings to return to. The overview/smlm thing is a pretty weighted case, but say I do a multicolor stack in between the overview and SMLM to decide whether a ROI is worth SMLM, which settings do we want the spoolcontroller to revert to each time?